File: crates/sl-frontend/src/lib/commands/accounts.ts
import { invoke } from '@tauri-apps/api/core';
import { ToastError } from '@/components/toasters';
import { PlayerAccounts } from '../types/account';

export const getAccounts = async () => {
	try {
		return await invoke<PlayerAccounts>('accounts_get');
	} catch (e) {
		ToastError(`${e}`);
	}
}

export const setCurrentAccount = async (name: string) => {
	try {
		await invoke('accounts_set_current', { name: name });
	} catch (e) {
		ToastError(`${e}`);
	}
}

export const createOfflineAccount = async (name: string) => {
	try {
		await invoke('accounts_create_offline', { name: name })
	} catch (e) {
		ToastError(`${e}`);
	}
}

export const removeAccount = async (name: string) => {
	try {
		await invoke('accounts_remove', { name: name })
	} catch (e) {
		ToastError(`${e}`)
	}
}

File: crates/sl-frontend/src-tauri/src/commands/accounts.rs
use sl_core::launcher::player_accounts::{add_account, remove_account, set_current_account, PlayerAccounts};
use sl_player::PlayerData;

#[tauri::command]
pub fn accounts_get() -> Result<PlayerAccounts , String> {
    PlayerAccounts::load().map_err(|err| err.to_string())
}

#[tauri::command]
pub fn accounts_remove(name: &str) -> Result<(), String> {
    remove_account(name).map_err(|e| e.to_string())?;
    Ok(())
}

#[tauri::command]
pub fn accounts_set_current(name: String) -> Result<(), String> {
    set_current_account(name).map_err(|err| err.to_string())?;
    Ok(())
}

#[tauri::command]
pub fn accounts_create_offline(name: String) -> Result<(), String> {
    add_account(name, PlayerData::default()).map_err(|e| e.to_string())?;
    Ok(())
}


